app/models/topic_group.rb

```diff
@@ -52,7 +52,7 @@ def self.create_topic_group(user, topic_id, post_number, updated_group_ids)
     SQL
 
     query +=
-      "AND NOT(tag.group_id IN (:already_updated_groups))" unless updated_group_ids.length.zero?
+      "AND NOT(tag.group_id IN (:already_updated_groups))" if updated_group_ids.length.nonzero?
```

Comment:
I think maybe this should be `positive?` , `nonzero?` returns self if !0 and matches negatives, whereas we want a boolean condition and this should never be negative. Could also just be `if updated_group_ids.any?`

Reply:
Actually, I went for `if updated_group_ids.present?` :grin: